packages/vite/src/node/optimizer/optimizer.ts

```diff
@@ -200,11 +200,23 @@ export function createDepsOptimizer(
             try {
               debug?.(colors.green(`scanning for dependencies...`))
 
-              discover = discoverProjectDependencies(
-                devToScanEnvironment(environment),
-              )
-              const deps = await discover.result
-              discover = undefined
+              let deps: Record<string, string>
+              try {
+                discover = discoverProjectDependencies(
+                  devToScanEnvironment(environment),
+                )
+                deps = await discover.result
+                discover = undefined
+              } catch (e) {
+                environment.logger.error(
+                  colors.red(
+                    '(!) Failed to run dependency scan. ' +
+                      'Skipping dependency pre-bundling. ' +
+                      e.stack,
+                  ),
+                )
+                return
+              }
```

Comment:
This part of the change just adds some context to the error. It looks like
![image](https://github.com/user-attachments/assets/49c6a1af-badb-45a8-a813-30ecefa95b3f)